online-fishing-store/src/app/pages/main-page/main-page.component.ts

```diff
@@ -25,18 +25,19 @@ export class MainPageComponent implements OnInit {
   constructor(
     public filterServise: FilterService,
     public sortServise: SortService,
-    public productServise: ProductService
+    public productService: ProductService
     ) {}
 
-  public getProducts(){
-    this.productServise.getProducts().subscribe((products) => {
-      this.products = products;
-      this.іnitialProducts = products;
-    })
+  public getProducts(): void{
+     this.productService.products$.subscribe((products:Product[]) => {
+        this.products = this.sortServise.sortByFavorites(products);
+        this.іnitialProducts = products;
+        this.sortServise.sortByFavorites(products)
```

Comment:
I can't understand for what you use in this line this.sortServise.sortByFavorites(products) method which you already did above